File: 3396-valid-word/3396-valid-word.cpp
class Solution {
public:
    bool isValid(string word) {
        if (word.length() < 3) return false;

        bool isVowel = false;
        bool isConsonant = false;

        for (auto letter : word) {
            if (isdigit(letter)) {
                continue;
            } 
            
            if (isalpha(letter)) {
                char lower = tolower(letter);
                if (lower == 'a' || lower == 'i' || lower == 'u' || lower == 'e' || lower == 'o') {
                    isVowel = true;
                } else {
                    isConsonant = true;
                }
            } else {
                return false;
            }
        }

        return isVowel && isConsonant;
    }
};

File: 3396-valid-word/3396-valid-word.java
import java.util.*;

class Solution {
    public boolean isValid(String word) {
        if (word.length() < 3) return false;

        boolean isVowel = false;
        boolean isConsonant = false;

        for (char letter: word.toCharArray()) {
            if (Character.isLetter(letter)) {
                char lower = Character.toLowerCase(letter);
                if ("aiueo".indexOf(lower) != -1) {
                    isVowel = true;
                } else {
                    isConsonant = true;
                }
            } else if (!Character.isDigit(letter)) {
                return false;
            }
        }

        return isVowel && isConsonant;
    }
}